addons/web/static/src/views/relational_model.js

```diff
@@ -3258,8 +3258,22 @@ export class RelationalModel extends Model {
                 return makeContext([rootParams.context], {});
             },
         };
-        const state = this.root ? this.root.exportState() : this.initialRootState;
-        const newRoot = this.createDataPoint(this.rootType, rootParams, state);
+        const loadParams = Object.assign({}, rootParams);
+        let state;
+        if (this.root) {
+            state = this.root.exportState();
+            if (this.rootType === "list" && !("limit" in params)) {
+                if (this.root instanceof DynamicRecordList) {
+                    delete loadParams.limit;
+                }
+                if (willBeGrouped(loadParams)) {
+                    loadParams.limit = state.limitByGroup;
+                }
+            }
+        } else {
+            state = this.initialRootState;
+        }
```

Comment:
I'm wondering if we chose the correct strategy. This doesn't look right (nor does the listState stuff in Group above). We should talk again about that.